addons/barfiller/statics.lua

```diff
@@ -0,0 +1,844 @@
+
+    if not sub_job_visible then
+        box:append(player.job..' ')                       -- JOB
+    else
+        box:append(player.job..player.sub)                -- JOB(sub)
+    end
+
+    if main_level_visible then box:append(player.lvl) end -- Lv##
+    if exp_visible        then box:append(player.exp) end -- EXP 59,999/60,000
+    if tnl_visible        then box:append(player.tnl) end -- (exp to next level)
+    if percent_visible    then box:append(player.pct) end -- ##%
+    if rate_visible       then box:append(player.phr) end -- EXP/hr #.#k
+end
+
+-- Display helpful information
+function help()
+    windower.add_to_chat(8,_addon.name..' v'.._addon.version..': Command Listing')
+    windower.add_to_chat(8,'   (c)lear - Resets EXP counter')
+    windower.add_to_chat(8,'   (u)nload - Disables BarFiller')
+    windower.add_to_chat(8,'   (r)eload - Reloads BarFiller')
+end
+
+-- Corrects EXP Bar's position on Level Up
+-- Thanks to Byrth's PointWatch addon
+function exp_msg(val,msg)
+    local t = os.clock()
+    if msg == 8 or msg == 105 then
+        xp.registry[t] = (xp.registry[t] or 0) + val
+        xp.current = math.min(xp.current + val,55999)
+        if xp.current > xp.tnl then
+            xp.current = xp.current - xp.tnl
+        end
+        chunk_update = true
+    end
+end
+
+-- Calculate EXP earned per hour
+-- Thanks to Byrth's PointWatch addon
+function analyze_points_table(tab)
+    local t = os.clock()
+    local running_total = 0
+    local maximum_timestamp = 29
+    for ts,points in pairs(tab) do
+        local time_diff = t - ts
+        if t - ts > 600 then
+            tab[ts] = nil
+        else
+            running_total = running_total + points
+            if time_diff > maximum_timestamp then
+                maximum_timestamp = time_diff
+            end
+        end
+    end
+    
+    local rate
+    if maximum_timestamp == 29 then
+        rate = 0
+    else
+        rate = math.floor((running_total/maximum_timestamp)*3600)
+    end
+    
+    return rate
+end
+
+-- Calculate XP Bar Width
+function calc_exp_bar()
+    if xp.current > 0 and xp.total > 0 then
+        local calc = math.floor((xp.current / xp.total) * 468)
+        xp.rate = analyze_points_table(xp.registry)
+        return calc
+    end
+end
+
+-- Center the Bar & Text on Screen
+function position_images()
+    set_bg_pos_x(((windower_res_x / 2) - (get_bg_width() / 2)))
+    set_fg_pos_x(get_bg_pos_x() + 2)
+    set_rb_pos_x(get_bg_pos_x() + get_bg_width())
+    set_rb_pos_y(get_bg_pos_y() - 6)
+end
+
+function position_text()
+    set_box_pos_x(get_bg_pos_x() - 6)
+    set_box_pos_y(get_bg_pos_y() + 4)
+end
+
+-- When logging out, hides the player stats
+function hide()
+    background_bar:hide()
+    foreground_bar:hide()
+    rested_bonus:hide()
+    box:clear()
+    ready = false
+end
+
+function mog_house()
+    local mh = windower.ffxi.get_info().mog_house
+    if mh then rested_bonus:show() else rested_bonus:hide() end
+end
+
```

Comment:
Do you need any of the following functions? The benefit of the library was so you could get rid off all of these. The entire rest of the file could be removed, if I understand this correctly.